transcom/mymove
# [MB-17008] Add left zero padding to LOA L1 segment values to meet 6 char validation

## [Jira ticket](https://dp3.atlassian.net/browse/MB-17008)

## Summary

We received feedback from DFAS about the EDI 858 LOA segments we were sending that the LOA L1 segment should always be six characters in length.  While some of the LOAs in our dataset have six character values for the `LoaInstlAcntgActID` others can be as small as three digit numerals (stored as string types).

This PR uses the format string to ensure we're left padding with zeroes to exactly 6 characters of length everytime unless we get a null value from the database.  In the case of a null value we will omit the L1 segment from the EDI like other fields instead of say sending "000000".  There is already a test that nil values get omitted so I didn't add a test for that.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

The easiest way to validate this PR is using the new test harness data:

1. Go to the test harness data: http://officelocal:3000/testharness/list
2. Click **MakeMoveReadyForEDI**
3. Search for and retain the payment request number it should look like: `"payment_request_number": "6841-8898-1"`
4. Build the generate-payment-request-edi cmd with `make bin/generate-payment-request-edi` to ensure you've compiled the latest changes.
5. Generate an EDI locally for the retained payment request number `bin/generate-payment-request-edi --payment-request-number <payment_request_number>`
6. Inspect the FA2 lines of the EDI, they should all contain the same data so verifying the first section is fine. You should see that the L1 value is now "000102" instead of "102", which is the value in the lines_of_accounting database table.
 
### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots

<img width="280" alt="Screenshot 2023-09-21 at 6 16 50 PM" src="https://github.com/transcom/mymove/assets/52669884/cb2ea70b-9540-4367-af74-84174caa8455">
